### PR TITLE
chore(TDP-5785) : modify suggest column endpoint

### PR DESCRIPTION
**Pull request by tlnd-jbrun**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5785

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java [hunk 1]
```diff
@@ -38,6 +38,21 @@
      */
     Stream<Suggestion> score(Stream<ActionDefinition> actions, ColumnMetadata column);
 
+    /**
+     * <p>
+     * Scores one <code>action</code> for a given <code>column</code>. Each suggestion is ranked from 0 to 1
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java]: Not sure do understand what you mean by "ranked from 0 to 1" as rules may add/subtract any integer to the score.

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java [hunk 2]
```diff
@@ -708,15 +708,17 @@ public GenericParameter dynamicParams(
         }
 
         // look for all actions applicable to the column type
-        final Stream<Suggestion> suggestions =
-                suggestionEngine.score(actionRegistry.findAll().parallel().filter(am -> am.acceptField(column)), column);
-        return suggestions //
+        return actionRegistry.findAll() //
+                .parallel() //
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java]: `parallel()` has a side effect described here -> https://dzone.com/articles/think-twice-using-java-8

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SimpleSuggestionEngine.java [hunk 3]
```diff
@@ -40,15 +40,27 @@
     @Override
     public Stream<Suggestion> score(Stream<ActionDefinition> actions, ColumnMetadata column) {
         return actions.map(actionMetadata -> { //
-                    int score = 0;
-                    for (SuggestionEngineRule rule : rules) {
-                        score += rule.apply(actionMetadata, column);
-                    }
-                    return new Suggestion(actionMetadata, score);
-                }) //
+            int score = 0;
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SimpleSuggestionEngine.java]: Why don't you use `score(ActionDefinition actionDef, ColumnMetadata column)` in the map iso. duplication code?

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java [hunk 4]
```diff
@@ -38,6 +38,21 @@
...
+     * <p>
+     * This method only operates on {@link ColumnMetadata}, meaning it can <b>not</b> decide based on content, only
+     * based on metadata.
+     * </p>
+     *
+     * @param action an {@link ActionDefinition action} to be scored.
+     * @param column The {@link ColumnMetadata column information} to be used to rank actions.
+     * @return A ordered collection of {@link Suggestion suggestions}.
+     */
+    Suggestion score(ActionDefinition action, ColumnMetadata column);
```
Comment (fhuaulme) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java]: I'm not sure to understand the use case of having a Suggestion out of a single ActionDefinition, can you elaborate in the PR and/or the Jira description?

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java (hunk 2 above)
Comment (tlnd-jbrun) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/TransformationService.java]: :+1:

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SimpleSuggestionEngine.java (hunk 3 above)
Comment (tlnd-jbrun) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SimpleSuggestionEngine.java]: :+1:

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java (hunk 1 above)
Comment (tlnd-jbrun) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java]: :+1:

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java (hunk 4 above)
Comment (fhuaulme) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/api/transformer/suggestion/SuggestionEngine.java]: (discussed offline and ok).

**Pull request merged**